### DISCUSSION TITLE
Enable mDNS Discovery but not searching for MQTT broker?

### DISCUSSION THREAD

**Discussion by probonopd** [arendst/Tasmota#13023] [Q&A] [answered]
Is it possible to announce the HTTP interface of Tasmota over mDNS (Zeroconf) but **not** have it search for a MQTT broker using mDNS?

Use case:
I would like my Tasmota devices to be easily accessible over the network without having to deal with IP addresses. But I don't want MQTT broker autodiscovery.

The documentation for `SetOption55` seems to mix these two things.

Is 

```
SetOption55
SetOption3 0
```

the only way to do this? (It entirely disables MQTT, which is not exactly what I want though.)

**Comment by barbudor** [comment, marked as answer]
I haven't played with that myself but looking at `my_user_config.h` (as an extensive catalog of almost all possible options and features), I ca n see:
```
// -- mDNS ----------------------------------------
//#define USE_DISCOVERY                            // Enable mDNS for the following services (+8k code or +23.5k code with core 2_5_x, +0.3k mem)
  #define WEBSERVER_ADVERTISE                    // Provide access to webserver by name <Hostname>.local/
  #define MQTT_HOST_DISCOVERY                    // Find MQTT host server (overrides MQTT_HOST if found)
```

So I would imagine that the following lines in your `user_configçoverride.h` would achieve it:
```
#define USE_DISCOVERY                            // Enable mDNS for the following services (+8k code or +23.5k code with core 2_5_x, +0.3k mem)
#define WEBSERVER_ADVERTISE                    // Provide access to webserver by name <Hostname>.local/
#undef MQTT_HOST_DISCOVERY                    // Find MQTT host server (overrides MQTT_HOST if found)
```

**Comment by Jason2866** [comment]
mDNS is a mess in general. Next issue the Arduino implementation still has bugs.
Thats the reason why mDNS is disabled by default (since that time weird issues in Tasmota are gone...
mDNS has never been tested with the Tasmota32 version. And the dev team will not do.

**Reply by sfromis** [reply]
I prefer something like having dnsmasq running (on my router) to resolve local domains without having to resort to mDNS.

**Reply by probonopd** [reply]
[DNS Service Discovery (DNS-SD)](http://www.dns-sd.org/) is imho awesome and easily one of the most under-appreciated technologies out there. Hope we can get it a first-class feature in Arduino ESP and Tasmota.

**Reply by Jason2866** [reply]
Will not be enabled in standard precompiled Tasmota versions since it needs a lot of flash space.
